tempodb/backend/gcs/gcs.go

```diff
@@ -67,12 +102,16 @@ func (rw *readerWriter) Write(ctx context.Context, meta *encoding.BlockMeta, bBl
 	defer src.Close()
 
 	w := rw.writer(ctx, util.ObjectFileName(blockID, tenantID))
-	defer w.Close()
 	_, err = io.Copy(w, src)
```

Comment:
if `io.Copy()` fails we never close the writer.  I guess you dropped the defer to catch the close error?  And maybe to force flush?

Reply:
That's correct.  Thoughts on best approach to this?

Reply:
Updated to close writer on other error.